lib/galaxy/objectstore/__init__.py

```diff
@@ -806,6 +981,12 @@ def _get_concrete_store_name(self, obj):
     def _get_concrete_store_description_markdown(self, obj):
         return self._call_method("_get_concrete_store_description_markdown", obj, None, False)
 
+    def _get_concrete_store_badges(self, obj):
+        return self._call_method("_get_concrete_store_badges", obj, [], False)
+
+    def _is_private(self, obj):
+        return self._call_method("_is_private", obj, ObjectNotFound, True)
```

Comment:
`ObjectNotFound ` leaks travels through the API, so as an unprivileged user you see for any deleted dataset
```json
{"err_msg":"objectstore, _call_method failed: _is_private on <galaxy.model.Dataset(212984) at 0x15c7ff760>, kwargs: {}","err_code":404001}
```
<img width="413" alt="Screenshot 2023-03-03 at 13 47 44" src="https://user-images.githubusercontent.com/6804901/222724180-50fd82c1-124d-4170-8988-a531fa202d8e.png">